src/plonk/lookup/prover.rs

```diff
@@ -414,16 +434,25 @@ impl<'a, C: CurveAffine> Committed<C> {
     ) {
         let permuted = self.permuted;
 
+        let active_rows = Polynomial::one_minus(pk.l_last.clone() + &pk.l_blind);
+
         let expressions = iter::empty()
-            // l_0(X) * (1 - z'(X)) = 0
+            // l_0(X) * (1 - z(X)) = 0
             .chain(Some(
                 Polynomial::one_minus(self.product_coset.clone()) * &pk.l0,
             ))
-            // z'(X) (a'(X) + \beta) (s'(X) + \gamma)
-            // - z'(\omega^{-1} X) (\theta^{m-1} a_0(X) + ... + a_{m-1}(X) + \beta) (\theta^{m-1} s_0(X) + ... + s_{m-1}(X) + \gamma)
+            // l_last(X) * (z(X)^2 - z(X)) = 0
+            .chain(Some(
+                (self.product_coset.clone() * &self.product_coset - &self.product_coset)
```

Comment:
```suggestion
                (self.product_coset.square() - &self.product_coset)
```

Reply:
There isn't a square method for Polynomial, let's address this another time.